specs/dotsigstore_bundle/pb/bundle.proto

```diff
@@ -0,0 +1,84 @@
+syntax = "proto3";
+package sigstore;
+
+// https://raw.githubusercontent.com/secure-systems-lab/dsse/9c813476bd36de70a5738c72e784f123ecea16af/envelope.proto
+import "envelope.proto";
+
+option go_package = "github.com/sigstore/xxx/bundle";
+
+enum HashAlgorithm {
+        SHA2_224 = 0;
+        SHA2_256 = 1;
+        SHA2_384 = 2;
+        SHA2_512 = 3;
+        SHA2_512_224 = 4;
+        SHA2_512_256 = 5;
+}
+
+message AttestationBlob {
+        // In JSON encoding this shall be base64 encoded.
+        bytes payload_hash = 1;
+        HashAlgorithm algorithm = 2;
+        // In JSON encoding this shall be base64 encoded.
+        bytes signature = 3;
+}
+
+// RekorEntry is a separate type, capturing all the details required
+// from Rekor to reconstruct an entry, given that the payload is provided
+// via other means.
+// This type can easily be created from the existing response from Rekor.
+// Future iterations could rely on Rekor returning the minimal set of
+// attributes (excluding the payload) that are required for verifying the
+// SET.
+message RekorEntry {
+        int64 log_index = 1;
+        string log_id = 2;
```

Comment:
Note that the signature is over the virtual index, as returned by rekor
https://github.com/sigstore/rekor/blob/ae429bbca05278bbeac3e49621bb7a791ce12d4e/pkg/api/entries.go#L220-L221

it should NOT be used for inclusion proof verification. that must use the leaf index in the shard.

Reply:
Yep, log ID is the sha256 digest of the PKIX-encoded public key of the log. We could also go with the OID approach that RFC9162 now recommends at some point

Reply:
The `log id` is returned as a string from Rekor today. To comply with RFC9162/OID we should change the data type to `bytes`. I would say this is fine, as we can just put the UTF8 encoding of the string returned from Rekor in there.
With this Rekor could change to an OID approach later, and this spec stay as-is.

Reply:
The downside with storing the string as `bytes` is that it will be base64 encoded when using JSON: (base64(hex(sha256(DER(public key)))).

Reply:
I have a preference to avoid `bytes` as much as possible, because it is unexpected that they're base64 encoded in JSON.

Reply:
I'm a little confused. Why would we re-encode the string we get from Rekor rather than decoding it and storing it as bytes? That avoids the base64+hex trap

Reply:
> I'm a little confused. Why would we re-encode the string we get from Rekor rather than decoding it and storing it as bytes?

You are thinking about this sentence, right: `as we can just put the UTF8 encoding of the string returned from Rekor in there`? 

Yes, you are right, we can decode the real bytes and put them in the `bytes` field. I was probably just not thinking clearly when I wrote that sentence.